iree/compiler/Dialect/LinalgExt/Transforms/ConvertToLoops.cpp

```diff
@@ -119,23 +124,166 @@ struct BubbleSortConversion : public OpRewritePattern<linalg_ext::SortOp> {
   }
 };
 
+//===----------------------------------------------------------------------===//
+// ScatterOp
+//===----------------------------------------------------------------------===//
+
+struct ScatterScalarConversion
+    : public OpRewritePattern<linalg_ext::ScatterOp> {
+  using OpRewritePattern<linalg_ext::ScatterOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg_ext::ScatterOp op,
+                                PatternRewriter& rewriter) const final {
+    if (!op.hasBufferSemantics()) return failure();
+
+    auto updates = op.getInputOperand(0);
```

Comment:
Nit: I think the `scatter` op has `updates()`, `indices()` and `original()` methods to do this. We could use that

Reply:
Good point, let me add the methods to linalg_ext.scatter op!

Reply:
I also added more methods, like `getUpdateSliceRank`, `isScalarUpdate`.